consul_exporter.go

```
@@ -75,10 +77,21 @@ type Exporter struct {
 }
 
 // NewExporter returns an initialized Exporter.
-func NewExporter(uri, kvPrefix, kvFilter string, healthSummary bool) *Exporter {
+func NewExporter(uri, kvPrefix, kvFilter string, healthSummary bool) (*Exporter, error) {
+	// parse uri to extract scheme
+	if !strings.Contains(uri, "://") {
+		uri = "http://" + uri
+	}
+
+	u, _ := url.Parse(uri)
```

Please let's not ignore the err.